R/testSingleVarianceComponent.R

```diff
@@ -25,7 +25,7 @@ testSingleVarianceComponent <- function(nullmod, varCompName, covMatList, group.
 
 	test.stat <- -2*(nullmod.noVarComp$logLik - nullmod$logLik)
 
-	pval <- 0.5*pchisq(test.stat, df = length(nullmod$varComp), lower.tail = FALSE) + 0.5*pchisq(test.stat, df = length(nullmod$varComp) - 1, lower.tail = F)
+	pval <- 0.5*.pchisq_filter_extreme(test.stat, df = length(nullmod$varComp), lower.tail = FALSE) + 0.5*.pchisq_filter_extreme(test.stat, df = length(nullmod$varComp) - 1, lower.tail = F)
```

Comment:
@amstilp -- I believe, for this test, it's possible for the df=0 for the reduced model, in which case the actual p-value for that component is 0. I think this might be a problem for your .pchisq_filter_extreme function

Reply:
Are the p-values returned by pchisq always 0 if df=0?

Reply:
Unless the test-statistic == 0, then the p-value = 1. 
A chi-square df = 0 distribution is a point mass at 0

Reply:
Ok. I can change it to something like this:
```
.pchisq_filter_extreme <- function(...) {
    args <- list(...)
    # Note: stat is the squared test statistic.
    pval = pchisq(...)
    if ("df" %in% names(args) && args$df > 0) {
        pval[pval < .Machine$double.xmin] = .Machine$double.xmin
    }
    return(pval)
}
```

which should cover the df=0 case?

Reply:
@amstilp -- I think "df" is a required argument, so do you really need to check if it's in the list?

Reply:
Good point - I rearranged the code a bit and the first check is no longer necessary. Will fix.